### PR TITLE
(VDB-950) Write raw diffs before transforming

### PR DESCRIPTION
**Pull request by rmulhol**
- Raw field we can reference by FK for related data
- Enables replay for unwatched or mistransformed diffs

Modified the transformer interface to take a `diffID` instead of `blockNumber` + `blockHash` so that we're not writing the same data to two places. I imagine we'll want to use `headerID` either instead of or in addition to the `diffID` in an upcoming story, so lmk if there seems like an easier way to set that up.

Also using `goimports -w <dir>` to manage imports without manual intervention 🎉

### CODE REVIEW

**Comment by i-norden** [comment]
Haven't had time to review but is this going to be rebased ontop of the backfill work or vice versa? Do you think there will be conflicts with that work?

**Comment by rmulhol** [comment]
@i-norden happy to rebase this on top of the backfill stuff - will take a look at that PR by tonight

File: libraries/shared/watcher/storage_watcher.go [hunk 1]
```diff
@@ -73,21 +79,31 @@ func (storageWatcher StorageWatcher) Execute(diffsChan chan utils.StorageDiff, e
...
-	storageTransformer, ok := storageWatcher.getTransformer(diff)
+func (storageWatcher StorageWatcher) processRow(diffInput utils.StorageDiffInput) {
+	diffID, err := storageWatcher.StorageDiffRepository.CreateStorageDiff(diffInput)
+	if err != nil {
+		if err == repositories.ErrDuplicateDiff {
+			logrus.Warn("ignoring duplicate diff")
+			return
+		}
+		logrus.Warnf("failed to persist storage diff: %s", err.Error())
+		// TODO: bail? Should we continue attempting to transform a diff we didn't persist
```
Comment (m0ar) [inline libraries/shared/watcher/storage_watcher.go]: errrrr, tricky. That would invalidate the expected data model in the DB right? Can we take the time to at least retry?

Comment (m0ar) [reply libraries/shared/watcher/storage_watcher.go]: Like, what are the potential reasons for this failure?

File: pkg/datastore/postgres/repositories/storage_diff_repository.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	row := repository.db.QueryRowx(`INSERT INTO public.storage_diff
+		(hashed_address, block_height, block_hash, storage_key, storage_value) VALUES ($1, $2, $3, $4, $5)
+		ON CONFLICT DO NOTHING RETURNING id`, input.HashedAddress.Bytes(), input.BlockHeight, input.BlockHash.Bytes(),
+		input.StorageKey.Bytes(), input.StorageValue.Bytes())
+	err := row.Scan(&storageDiffID)
+	if err != nil && err == sql.ErrNoRows {
+		return 0, ErrDuplicateDiff
+	}
+	return storageDiffID, err
+}
```
Comment (m0ar) [inline pkg/datastore/postgres/repositories/storage_diff_repository.go]: :ok_hand: :ok_hand:

**Pull request merged**